modules/packages/HDF5_HL.chpl

```diff
@@ -3516,4 +3517,128 @@ module HDF5_HL {
       }
     }
   }
+
+  module HDF5_Chapel {
+    /*
+    // This record could be used to open and close an HDF5 file based on
+    // simple scoping rules.  It is currently not used.
+    record HDF5_file {
+      const filename: string;
+      const file_id: hid_t;
+
+      proc init(name: string, access: c_uint = H5F_ACC_RDONLY) {
+        filename = name;
+        file_id = H5Fopen(name.c_str(), access, H5P_DEFAULT);
+      }
+
+      proc deinit() {
+        H5Fclose(file_id);
+      }
+    }
+    */
+
+    /* Read the dataset named `dsetName` out of all HDF5 files in the
+       directory `dirname` with filenames that begin with `filenameStart`.
+       This will read the files in parallel with one task per locale in the
+       `locs` array.
```

Comment:
Seems like it `locs` contains duplictate locales then you can read using multiple tasks per locale. Because each task will have its own process. If so, how about reflect it in this chpldoc comment somehow?

Reply:
How about add something like "It will return a distributed array of ArrRec instances. Each instance will reside on the locale where the corresponding data was read."

What do you think whether this is the right behavior or the return thing should have all the arrays on the same locale?

Reply:
If `locs` contains duplicate locales they will refer to the same process, and it will cause undefined behavior within HDF5.  I'll add this to the comment.

I'll add a comment about the return type.

I think the returned thing being all on one locale or being distributed could both be the right behavior depending on how many arrays there are, how big they are, and how they will be accessed.